internal/app/feed/exporter_report.go

```diff
@@ -0,0 +1,342 @@
+package feed
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/SafetyCulture/iauditor-exporter/internal/app/api"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+// ReportExporter is an interface to export data feeds to CSV files
+type ReportExporter struct {
+	*SQLExporter
+	Logger       *zap.SugaredLogger
+	ExportPath   string
+	PreferenceID string
+	Format       []string
+	Mu           sync.Mutex
+}
+
+type ReportExportFormat struct {
+	PDF  bool
+	WORD bool
+}
+
+type Report struct {
```

Comment:
I assume report is being used to link an audit_id to a file name? Can we just store the full path in that case instead of the int? Seems a bit simpler.

Reply:
I used to have a `boolean` value instead of `int`. The benefit of `int` is that I use `-1` as error status so I can use it for the final reporting which is shown when the process is finished.